x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -191,10 +192,26 @@ export function trainedModelsRoutes(
             mlLog.debug(e);
           }
 
-          const body = filterForEnabledFeatureModels(result, getEnabledFeatures());
+          try {
+            if (withOriginJobCheck) {
+              for (const model of result) {
+                if (typeof model.metadata?.analytics_config?.id === 'string') {
+                  const jobExistsResult = await mlClient.getDataFrameAnalytics({
+                    id: model.metadata?.analytics_config?.id,
+                    size: 1,
+                  });
+                  model.origin_job_exists = jobExistsResult.count === 1;
+                }
+              }
+            }
+          } catch (e) {
+            // Swallow error to prevent blocking trained models result
```

Comment:
If one of the jobs has been deleted, it looks like following models in the list are having the action disabled even when the job exists.

Reply:
Fixed in 7c06abd24a23cdfb98b34fba27bfdce6aee3e1ad